be/src/olap/rowset/segment_v2/plain_page.h

```diff
@@ -0,0 +1,170 @@
+    }
+
+    Slice finish(const rowid_t page_first_rowid) override {
+        encode_fixed32_le((uint8_t *) &_buffer[0], _count);
+        encode_fixed32_le((uint8_t *) &_buffer[4], page_first_rowid);
+        return Slice(_buffer.data(),  plainPageHeaderSize + _count * SIZE_OF_TYPE);
+    }
+
+    void reset() override {
+        _count = 0;
+        _buffer.clear();
+        _buffer.resize(plainPageHeaderSize);
+    }
+
+    size_t count() const {
+        return _count;
+    }
+
+    // this api will release the memory ownership of encoded data
+    // Note:
+    //     release() should be called after finish
+    //     reset() should be called after this function before reuse the builder
+    void release() override {
+        uint8_t *ret = _buffer.release();
+        (void) ret;
+    }
+
+private:
+    faststring _buffer;
+    const PageBuilderOptions *_options;
+    size_t _count;
+    typedef typename TypeTraits<Type>::CppType CppType;
+    enum {
+        SIZE_OF_TYPE = TypeTraits<Type>::size
+    };
+};
+
+
+template<FieldType Type>
+class PlainPageDecoder : public PageDecoder {
+public:
+    PlainPageDecoder(Slice data) : _data(data),
+              _parsed(false),
+              _num_elems(0),
+              _page_first_ordinal(0),
+              _cur_idx(0) { }
+
+    Status init() override {
+        CHECK(!_parsed);
+
+        if (_data.size < plainPageHeaderSize) {
+            std::stringstream ss;
+            ss << "file corrupton: not enough bytes for header in PlainBlockDecoder ."
+                  "invalid data size:" << _data.size << ", header size:" << plainPageHeaderSize;
+            return Status::InternalError(ss.str());
+        }
+
+        _num_elems = decode_fixed32_le((const uint8_t *) &_data[0]);
+        _page_first_ordinal = decode_fixed32_le((const uint8_t *) &_data[4]);
+
+        if (_data.size != plainPageHeaderSize + _num_elems * SIZE_OF_TYPE) {
+            std::stringstream ss;
+            ss << "file corrupton: unexpected data size.";
+            return Status::InternalError(ss.str());
+        }
+
+        _parsed = true;
+
+        seek_to_position_in_page(0);
+        return Status::OK();
+    }
+
+    Status seek_to_position_in_page(size_t pos) override {
+        CHECK(_parsed) << "Must call init()";
+
+        if (PREDICT_FALSE(_num_elems == 0)) {
+            DCHECK_EQ(0, pos);
+            return Status::InternalError("invalid pos");
+        }
+
+        DCHECK_LE(pos, _num_elems);
+
+        _cur_idx = pos;
+        return Status::OK();
+    }
+
+    Status next_batch(size_t *n, ColumnVectorView *dst) override {
+        DCHECK(_parsed);
+
+        if (PREDICT_FALSE(*n == 0 || _cur_idx >= _num_elems)) {
+            *n = 0;
+            return Status::OK();
+        }
+
+        size_t max_fetch = std::min(*n, static_cast<size_t>(_num_elems - _cur_idx));
+        memcpy(dst->column_vector()->col_data(),
```

Comment:
should consider row_offset in ColumnVectorView here.
And How to process string and binary values?
It seems that you should store value length for string or binary data?